interview-questions.md

```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: 'this' is a designator used inside of an object or class in order to refer to the class or object one is inside of when calling a method or key in a key:value pair.
 
-  Researched answer:
+  Researched answer: I was sort of correct. It appears that 'this' refers to an object and when used inside of an object, it refers to the object itself. Outside of an object it either refers to the global object or is undefined.
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a Javascript library which contains code necessary for building and combining components. Compenents are pieces of code in React which can be built to display HTML and run JSX commands on an HTML page. You can use React to build websites or apps which are both scalable and modular due to the various components used in their construction.
 
-  Researched answer:
+  Researched answer: React is not only a library, but a free and open-source front-end Javascript library. It is based on components and can be used to build single-page, mobile, or server-rendered apps.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: Drawing a blank on this one.
 
-  Researched answer:
+  Researched answer: The constructor() method is required in a React class component because constructor handles state initialization and event handlers, which all affect how the component is rendered on the Document Object Model.
```

Comment:
Since React is always looking to display or _render_ something the render() method is the required lifecycle method in a React class component.